README.md

```diff
@@ -25,3 +25,105 @@ Hi, I'm your host, Dima ([Github](https://github.com/dkorolev), [LinkedIn](https
 * MapReduce, [video](https://www.youtube.com/watch?v=HR82uF7vQ0k), [slides](https://docs.google.com/presentation/d/19_Nk8XhmUZfNgdKNV6qL9pb5yhzhUPOp-LPZ8aCMeG0).
+
+# Educational Designs
+
+We are looking into working on designing a real-life system, or a few of them, in a meetup-orchestrated crowdsourced fashion.
+
+_[ This is the early stage effort, and the below is subject to change. In a way, it's an RFC. Also, it should probably be moved into a separate `.md` file. ]_
+
+Here are some core principles I would like to adhere to while doing so.
+
+1. **Distributed First**.
+
+   The theme of the meetup is system design, and I personally understand system design primarily in the context of distributed dataflows.
+
+   When designed properly, each component can fail, and the state machine that governs the system should remain logically unified yet physically distributed, so that it can correctly satisfy all the functional requirements and do its best to fit the non-functional ones.
+
+   Also, the distributed-first paradigm enables implementing individual components of the system in different programming languages, to then compare their performance in various cross-combinations.
+
+   A well-designed distributed system is a rare example of something that is far greater than the sum of its parts. The principles by which we design them should result in systems that illustrate this thesis.
+
+2. **Technology-Agnostic**.
+
+   Since the focus is on what happens between components, we should strive to build individual components in the least opinionated ways possible.
+
+   Language-agnostic comes first. Independent components should be able to effectively work together regardless of the programming language in which each of them is implemented. In fact, I expect most components to be implemented in more than one language.
+
+   Docker is a must have these days, whether we like it or not. Ideally, each component should be designed such that it can be built & run on bare metal, or packed into a Debian module or into a unikernel of its own, or even be coupled with other components into a single combined piece. Still, for the purposes of keeping the design clean, each component should come with a `Dockerfile`, so that a `docker compose` can be natively used to spin up and run & test the entire constellation.
+
+   In particular, if a certain component requires warmup, or has nontrivial dependencies, we implement these in a Docker-native way first, using health checks and additional containers so that the dependency graph is respected by any well-designed test.
+
+   I also want our components to be agnostic with respect to now-common standard building blocks. There should be no "Database" or "Cache" or "Message Queue" or "Leader Elections" components, and there should be no Postgres/MySQL/Mongo, Redis/Memcached, Kafka/RabbitMQ, or Consul/etcd ones. Instead, each component should serve a specific well-articulated logical function.
```

Comment:
I didn't get it. Does it mean that we don't want to implement common solutions "from scratch"? Or that we don't want to allot all space of any component for say DB without any additional logic?

Reply:
:+1: 014c279

Reply:
:+1: 014c279